addon/components/frost-checkbox.js

```diff
@@ -1,34 +1,50 @@
+/**
+ * Component definition for the frost-checkbox component
+ */
 import Ember from 'ember'
-const {
-  Component,
-  get,
-  isEmpty,
-  run,
-  set,
-  typeOf
-} = Ember
+const {Component, get, isEmpty, run, set, typeOf} = Ember
 import PropTypeMixin, {PropTypes} from 'ember-prop-types'
+
 import layout from '../templates/components/frost-checkbox'
 
 export default Component.extend(PropTypeMixin, {
-  // == Properties ============================================================
+  // == Dependencies ==========================================================
+
+  // == Keyword Properties ====================================================
 
-  classNames: ['frost-checkbox'],
   classNameBindings: ['size'],
+  classNames: ['frost-checkbox'],
   layout,
 
+  // == PropTypes =============================================================
+
+  /**
+   * Properties for this component. Options are expected to be (potentially)
+   * passed in to the component. State properties are *not* expected to be
+   * passed in/overwritten.
+   */
```

Comment:
I thought, since it's not common knowledge about the separation of 'options' and 'state' that it would be beneficial. If you think everyone already knows about that (which would surprise me, but it's been known to happen) I'm fine with removing it.